SharedRules.DotSettings

```diff
@@ -0,0 +1,1186 @@
+	
+	
+	
+	
+	<s:Boolean x:Key="/Default/PatternsAndTemplates/StructuralSearch/Pattern/=04714CEE450EE047AEDF4094CA95BEA7/CustomPatternPlaceholder/=statements1/@KeyIndexDefined">True</s:Boolean>
+	<s:String x:Key="/Default/PatternsAndTemplates/StructuralSearch/Pattern/=04714CEE450EE047AEDF4094CA95BEA7/CustomPatternPlaceholder/=statements1/Properties/=Maximal/@EntryIndexedValue">-1</s:String>
+	<s:String x:Key="/Default/PatternsAndTemplates/StructuralSearch/Pattern/=04714CEE450EE047AEDF4094CA95BEA7/CustomPatternPlaceholder/=statements1/Properties/=Minimal/@EntryIndexedValue">0</s:String>
+	<s:String x:Key="/Default/PatternsAndTemplates/StructuralSearch/Pattern/=04714CEE450EE047AEDF4094CA95BEA7/CustomPatternPlaceholder/=statements1/Type/@EntryValue">StatementPlaceholder</s:String>
+	<s:Boolean x:Key="/Default/PatternsAndTemplates/StructuralSearch/Pattern/=04714CEE450EE047AEDF4094CA95BEA7/CustomPatternPlaceholder/=statements2/@KeyIndexDefined">True</s:Boolean>
+	<s:String x:Key="/Default/PatternsAndTemplates/StructuralSearch/Pattern/=04714CEE450EE047AEDF4094CA95BEA7/CustomPatternPlaceholder/=statements2/Properties/=Maximal/@EntryIndexedValue">-1</s:String>
+	<s:String x:Key="/Default/PatternsAndTemplates/StructuralSearch/Pattern/=04714CEE450EE047AEDF4094CA95BEA7/CustomPatternPlaceholder/=statements2/Properties/=Minimal/@EntryIndexedValue">0</s:String>
+	<s:String x:Key="/Default/PatternsAndTemplates/StructuralSearch/Pattern/=04714CEE450EE047AEDF4094CA95BEA7/CustomPatternPlaceholder/=statements2/Type/@EntryValue">StatementPlaceholder</s:String>
+	
+	
+	
+	
+	<s:String x:Key="/Default/PatternsAndTemplates/StructuralSearch/Pattern/=04714CEE450EE047AEDF4094CA95BEA7/LanguageName/@EntryValue">CSHARP</s:String>
+	<s:Boolean x:Key="/Default/PatternsAndTemplates/StructuralSearch/Pattern/=04714CEE450EE047AEDF4094CA95BEA7/MatchCatchClauseWithoutExceptionFilter/@EntryValue">False</s:Boolean>
+	<s:String x:Key="/Default/PatternsAndTemplates/StructuralSearch/Pattern/=04714CEE450EE047AEDF4094CA95BEA7/SearchPattern/@EntryValue">for (int $index$ = $expr$; $index$ &lt; $expr2$; ++$index$)&#xD;
+{&#xD;
+	$statements1$&#xD;
+	if($ifclause$)&#xD;
+	{&#xD;
+		$index$ = $expr3$;&#xD;
+	}&#xD;
+	$statements2$&#xD;
+}</s:String>
+	<s:String x:Key="/Default/PatternsAndTemplates/StructuralSearch/Pattern/=04714CEE450EE047AEDF4094CA95BEA7/Severity/@EntryValue">WARNING</s:String>
+	<s:String x:Key="/Default/PatternsAndTemplates/StructuralSearch/Pattern/=04714CEE450EE047AEDF4094CA95BEA7/SuppressionKey/@EntryValue">DontChangeLoopVar</s:String>
+	<s:Boolean x:Key="/Default/PatternsAndTemplates/StructuralSearch/Pattern/=0A70B2667067AE4E820D441F476A5DB7/@KeyIndexDefined">True</s:Boolean>
+	<s:String x:Key="/Default/PatternsAndTemplates/StructuralSearch/Pattern/=0A70B2667067AE4E820D441F476A5DB7/Comment/@EntryValue">Don't make explicit comparisons to true or false (SDCS-1314)</s:String>
```

Comment:
Why don't you use built-in inspection?
![image](https://user-images.githubusercontent.com/4506967/73163419-069dd080-4101-11ea-9add-4976d0a5cc6e.png)

Reply:
it works only with constants, it allows comparing expressions with true or false

Reply:
yeap, it looks like built-in validation works only with smth like this ```expression == true``. So your custom pattern partially duplicates 'Redundant bool comparison'. I think it makes sense to disable this rule and keep Custom pattern.

Reply:
done